### PR TITLE
[nodejs] use compression to shorten chartUrl

### PR DESCRIPTION
**Pull request by chejennifer**
Sometimes chartUrl that is returned in the response is too long to send a GET request to. Use compression to shorten the URL

For example, "Which industries have grown the most in California" chartUrls:
current (5810 characters): https://screenshot.googleplex.com/4WUDYVS5N2MQsVj
with compression (1513 characters): https://screenshot.googleplex.com/8HAWVmq7Yef9BQE

### CODE REVIEW

File: static/nodejs_server/utils.ts [hunk 1]
```diff
@@ -139,9 +148,13 @@ export function getChartUrl(
       if (!paramVal) {
         return;
       }
```
Comment (pradh) [inline static/nodejs_server/utils.ts:150]: So, we're compressing each value independently?  Is there value in compressing stuff like place, enclosed_place_type.

Or how about just compressing the whole thing in one shot and passing `arg=<base64-encoded-stuff>`?

File: static/nodejs_server/utils.ts [hunk 2]
```diff
@@ -139,9 +148,13 @@ export function getChartUrl(
...
-      url += `${idx === 0 ? "" : "&"}${paramKey}=${paramVal}`;
+      let compressedVal = zlib.deflateSync(paramVal).toString("base64");
```
Comment (pradh) [inline static/nodejs_server/utils.ts:151]: How about having a helper for `compress` and `decompress` that wraps this COMPRESSED_VAL_ENCODING replacement?

Also, maybe have a simple unit-test that compresses, decompresses and confirms they match?

File: static/src/server.ts [hunk 3]
```diff
@@ -616,21 +621,35 @@ app.get("/nodejs/query", (req: Request, res: Response) => {
...
+    let val = req.query[key] as string;
+    // Convert all the manually escaped characters back to their original
+    // characters.
+    for (const c in COMPRESSED_VAL_ENCODING) {
+      val = val.replaceAll(COMPRESSED_VAL_ENCODING[c], c);
+    }
+    // decompress the value
+    val = zlib.inflateSync(Buffer.from(val, "base64")).toString();
+    decompressedReqVals[key] = val;
+  }
```
Comment (pradh) [inline static/src/server.ts:636]: Put in decompress wrapper?

**Review by pradh — COMMENTED** [review]
Thanks!   Are we not dropping the names?

File: static/nodejs_server/utils.ts (hunk 1 above)
Comment (chejennifer) [reply static/nodejs_server/utils.ts:152]: ah, good idea, updated

File: static/nodejs_server/utils.ts (hunk 2 above)
Comment (chejennifer) [reply static/nodejs_server/utils.ts]: done

File: static/src/server.ts (hunk 3 above)
Comment (chejennifer) [reply static/src/server.ts]: done

**Comment by chejennifer** [comment]
> Thanks! Are we not dropping the names?

Thanks for the review! I saw that just by using compression, the URLs are shortened by a lot, so wanted to get this in first and then dropping names can be in a future PR?

**Review by pradh — APPROVED** [review]

**Comment by pradh** [comment]
> > Thanks! Are we not dropping the names?
> 
> Thanks for the review! I saw that just by using compression, the URLs are shortened by a lot, so wanted to get this in first and then dropping names can be in a future PR?

Sounds good!  Thanks for the changes and test.

**Pull request merged**